src/App.tsx

```diff
@@ -32,28 +32,28 @@ const client = new ApolloClient({
   cache: new InMemoryCache(),
 });
 
-type Post = {
-  id: number;
-  title: string;
-  body: string;
-};
-
-export const PostItem = (post: Post) => (
+export const PostItem = (post: PostItemInterface) => (
   <div className="bg-white rounded-md px-2 py-1 mt-4">
-    <h2>{post.title}</h2>
-    <p className="text-dark-body text-sm m-0">{post.body}</p>
-  </div>
-);
-
-const NewPostItem = (post: Post) => (
-  <div key={post.id} className="bg-secondary rounded-md px-2 py-1 mt-4">
-    <h2>{post.title}</h2>
-    <p className="text-dark-body text-sm m-0">{post.body}</p>
+    <h2>{post?.title}</h2>
+    <p className="text-dark-body text-sm m-0">{post?.body}</p>
   </div>
 );
 
 export const Main = () => {
   const { loading, error, data } = useQuery(GET_POSTS);
+  const [showCreateItem, setShowCreateItem] = useState<boolean>(false)
```

Comment:
Can you propose a different way on how to show the form conditionally that would not involve a _state_ in the `Main` component?

Hint: We are looking for performance.